src/main/java/org/jabref/gui/BasePanel.java

```diff
@@ -539,11 +507,12 @@ public void updateEntryEditorIfShowing() {
         }
     }
 
+    /**
+     * Put an asterisk behind the filename to indicate the database has changed.
```

Comment:
Can you please investigate if this method is still really necessary. In principle, it should be enough to add listeners to the state manager to be notified about db changes. This would also fix the strange flow of information (from the base panel to the main frame).

Reply:
I intensivly investigated this issue. I believe this method or something like it is still necessary until we can bind some property to an undomanager, which can provide a dirty flag for each library file.